### DISCUSSION TITLE
How to implement on existing system with auto incrementing IDs?

### DISCUSSION THREAD

**Discussion by marlocorridor** [hirethunk/verbs#124] [Q&A]
Hi, saw this package on Laravel News.
This package could help me develop the employee Points management feature of our system.

As I understand on the [documentation](https://verbs.thunk.dev/docs/technical/ids), we need to convert IDs to Snowflakes.

🤔 Could we create only the related models for this feature to use snowflakes?
Could you share your experience so I could ease my hesitation. 😆 

Thanks a lot!

Using: laravel nova, livewire

**Comment by nick-potts** [comment]
Only the internal ids/states need to use the snowflakes, your regular eloquent models can use whatever you want.

**Reply by marlocorridor** [reply]
So this is ok? 🤔 

```php
class EmployeeInitializePoints extends Event
{
    #[StateId(EmployeeState::class)]
    public ?int $state_id = null; // the snowflake ID

    public ?int $customer_id = null; // my current auto inc IDs 
}
```

**Reply by nick-potts** [reply]
Looks fine to me

**Reply by tsterker** [reply]
See also this blog post by @inxilpro:

> [Models in Verbs](https://cmorrell.com/models-in-verbs)
